From 276c2f1334345071eaeef9aea87c96cdd199f8f0 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sun, 11 Jun 2023 19:56:59 +0200
Subject: [PATCH] Fix kdeconnect_open desktop file type

It's an application, not a service

Also don't make it executable, it's not needed

BUG: 424782
(cherry picked from commit 7569a6ceffc6fc169186e401d33ce1f2049ec5e4)
---
 data/CMakeLists.txt                  | 2 +-
 data/org.kde.kdeconnect_open.desktop | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 5d673f0ab..6d3621f29 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -1,6 +1,6 @@
 install (FILES "kdeconnect.contract" DESTINATION ${KDE_INSTALL_DATADIR}/contractor/)
 install (FILES "kdeconnect-thunar.desktop" DESTINATION ${KDE_INSTALL_DATADIR}/Thunar/sendto/)
-install(PROGRAMS org.kde.kdeconnect_open.desktop DESTINATION ${KDE_INSTALL_APPDIR})
+install(FILES org.kde.kdeconnect_open.desktop DESTINATION ${KDE_INSTALL_APPDIR})
 install (FILES "kdeconnect-dde.desktop" DESTINATION ${KDE_INSTALL_DATADIR}/deepin/dde-file-manager/oem-menuextensions/)
 
 if(NOT SAILFISHOS)
diff --git a/data/org.kde.kdeconnect_open.desktop b/data/org.kde.kdeconnect_open.desktop
index 03de93ef6..110dc2f67 100644
--- a/data/org.kde.kdeconnect_open.desktop
+++ b/data/org.kde.kdeconnect_open.desktop
@@ -1,4 +1,5 @@
 [Desktop Entry]
+Type=Application
 GenericName=Open on connected device via KDE Connect
 GenericName[ar]=افتح في الجهاز المتصل عبر «كِيدِي المتّصل»
 GenericName[az]=Connectd cihazında KDE Connect ilə açın
@@ -128,7 +129,6 @@ Comment[zh_TW]=使用 KDE 連線於連線裝置中開啟
 MimeType=application/octet-stream;
 Exec=kdeconnect-handler --open %U
 Icon=kdeconnect
-Type=Service
 Terminal=false
 Categories=Qt;KDE;Network;
 X-KDE-FormFactor=desktop;tablet;handset;
-- 
GitLab

